JotunnLib/Managers/DungeonManager.cs

```diff
@@ -0,0 +1,387 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using HarmonyLib;
+using Jotunn.Entities;
+using Jotunn.Managers.MockSystem;
+using Jotunn.Utils;
+using UnityEngine;
+
+namespace Jotunn.Managers
+{
+    /// <summary>
+    ///    Manager for handling all custom data added to the game related to creatures.
+    /// </summary>
+    public class DungeonManager : IManager
+    {
+        private static DungeonManager _instance;
+
+        /// <summary>
+        ///     The singleton instance of this manager.
+        /// </summary>
+        public static DungeonManager Instance => _instance ??= new DungeonManager();
+
+        /// <summary>
+        ///     Event that gets fired after the vanilla <see cref="DungeonDB"/> has loaded rooms.  Your code will execute
+        ///     every time a main scene is started (on joining a game). <br />
+        ///     If you want to execute just once you will need to unregister from the event after execution.
+        /// </summary>
+        public static event Action OnVanillaRoomsAvailable;
```

Comment:
Best to provide a OnRoomsRegistered event, too

Reply:
@MSchmoecker Do I need to do anything other than just adding the below line?

`public static event Action OnRoomsRegistered;`

Reply:
the event has to be invoked like `OnVanillaRoomsAvailable` with `InvokeOnVanillaRoomsAvailable` after the rooms have been registered